### PR TITLE
Feat(scoped variables): floater suggestions

### PR DESCRIPTION
**Pull request by AbhishekA1509**
# Description
Floating Suggestions widget, to show list of available variables:
Draggable collapsed state:
Screenshot 2023-09-12 at 10 40 06 AM

Draggable Expanded state:
Screenshot 2023-09-12 at 10 40 53 AM

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Unit Test coverage of around 90%

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/css/formulae.scss [hunk 1]
```diff
@@ -145,7 +145,7 @@ $color-suffix: 00;
 }
 
 // border-radius
-@for $i from 0 through 16 {
+@for $i from 0 through 48 {
```
Comment (vivek-devtron) [inline src/css/formulae.scss:148]: Please revert this and create class of br-48

Comment (AbhishekA1509) [reply src/css/formulae.scss]: Done

File: src/components/common/FloatingVariablesSuggestions/SuggestionItem.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+export default function SuggestionItem({
+    variableName,
+    description,
+    variableValue,
+    isRedacted,
+    highlightText,
+}: SuggestionsItemProps) {
+    const [triggerCopy, setTriggerCopy] = useState<boolean>(false)
+
+    const handleCopyTrigger = useCallback(() => setTriggerCopy(true), [])
```
Comment (vivek-devtron) [inline src/components/common/FloatingVariablesSuggestions/SuggestionItem.tsx:17]: we can avoid using useCallback here

File: src/components/common/FloatingVariablesSuggestions/Suggestions.tsx [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    const [clearSearch, setClearSearch] = useState<boolean>(false)
+    const [highlightText, setHighlightText] = useState<string>('')
+
+    const enableSearch = !loading && !error && !!variables?.length
+
+    useEffect(() => {
+        setSuggestions(variables)
+    }, [variables])
+
+    const onSearch = useCallback(
```
Comment (vivek-devtron) [inline src/components/common/FloatingVariablesSuggestions/Suggestions.tsx:24]: We can avoid useCallback here

File: src/components/common/FloatingVariablesSuggestions/Suggestions.tsx [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+            setHighlightText(text.trim())
+        },
+        [variables],
+    )
+
+    const handleClearSearch = useCallback(() => {
+        setClearSearch(!clearSearch)
+    }, [clearSearch])
+
+    const renderClearSearchButton = useCallback(
```
Comment (vivek-devtron) [inline src/components/common/FloatingVariablesSuggestions/Suggestions.tsx:42]: same here

File: src/components/common/FloatingVariablesSuggestions/Suggestions.tsx [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+                    variable.variableName.toLowerCase().includes(text.trim().toLowerCase()) ||
+                    variable.shortDescription?.toLowerCase().includes(text.trim().toLowerCase()),
+            )
+            setSuggestions(filteredSuggestions)
+            setHighlightText(text.trim())
+        },
+        [variables],
+    )
+
+    const handleClearSearch = useCallback(() => {
```
Comment (vivek-devtron) [inline src/components/common/FloatingVariablesSuggestions/Suggestions.tsx:38]: same here

Comment (AbhishekA1509) [reply src/components/common/FloatingVariablesSuggestions/Suggestions.tsx]: Done

File: src/components/common/FloatingVariablesSuggestions/Suggestions.tsx (hunk 4 above)
Comment (AbhishekA1509) [reply src/components/common/FloatingVariablesSuggestions/Suggestions.tsx]: Done

File: src/components/common/FloatingVariablesSuggestions/Suggestions.tsx (hunk 3 above)
Comment (AbhishekA1509) [reply src/components/common/FloatingVariablesSuggestions/Suggestions.tsx]: Done

File: src/components/common/FloatingVariablesSuggestions/SuggestionItem.tsx (hunk 2 above)
Comment (AbhishekA1509) [reply src/components/common/FloatingVariablesSuggestions/SuggestionItem.tsx]: Done

File: src/components/common/FloatingVariablesSuggestions/Suggestions.tsx (hunk 3 above)
Comment (AbhishekA1509) [reply src/components/common/FloatingVariablesSuggestions/Suggestions.tsx]: Done

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**